starknet_py/net/full_node_client.py

```diff
@@ -536,19 +573,43 @@ def _create_broadcasted_txn(transaction: AccountTransaction) -> dict:
     }
 
 
-def _create_broadcasted_declare_properties(transaction: Declare) -> dict:
+def _create_broadcasted_declare_properties(
+    transaction: Union[Declare, DeclareV2]
+) -> dict:
+    if isinstance(transaction, DeclareV2):
+        return _create_broadcasted_declare_v2_properties(transaction)
+
     contract_class = cast(Dict, DeclareSchema().dump(obj=transaction))["contract_class"]
     declare_properties = {
         "contract_class": {
-            "program": contract_class["program"],
             "entry_points_by_type": contract_class["entry_points_by_type"],
             "abi": contract_class["abi"],
```

Comment:
I know it was like this, but should `abi` key be present even if the contract_class["abi"] is None?

Reply:
True, it should be optional, will change that.